fsspec/mapping.py

```diff
@@ -34,16 +37,18 @@ class FSMap(MutableMapping):
 
     def __init__(self, root, fs, check=False, create=False, missing_exceptions=None):
         self.fs = fs
-        self.root = fs._strip_protocol(root).rstrip(
-            "/"
-        )  # we join on '/' in _key_to_str
+        self.root = fs._strip_protocol(root).rstrip("/")
+        self._root_key_to_str = fs._strip_protocol(os.path.join(root, "x"))[:-1]
         if missing_exceptions is None:
-            missing_exceptions = (
+            missing_exceptions = [
                 FileNotFoundError,
                 IsADirectoryError,
                 NotADirectoryError,
-            )
-        self.missing_exceptions = missing_exceptions
+            ]
+            if platform.system() == "Windows":
+                # see: https://bugs.python.org/issue43095
+                missing_exceptions.append(PermissionError)
```

Comment:
Actually, this is problematic. The simple problem is, that windows users will of course use this API with FSs other than local. Secondly, there may be real permission errors - should they really appear missing?

Reply:
That makes sense. I'll revert the commit and skip the relevant test on windows.